doc/kb_sdk_make_ui.md

```diff
@@ -0,0 +1,169 @@
+                        "display": "Do not prune",
+                        "id": "none",
+                        "ui_name": "Do not prune"
+                    }
+                ]
+            }
+        },
+```
+
+There are many additional interface options available. One of the best ways to discover them is to look at 
+[this gallery](https://narrative.kbase.us/narrative/ws.23109.obj.1) which contains a variety of KBase apps along with 
+the spec file that generated the user interface.
+
+[\[Back to top\]](#top)
+
+#### B . Configure passing variables from Narrative Input to SDK method.
+
+In the 'behavior' section of the spec.json, the output of the user interface is mapped to input to your function.
+If you have maintained a consistent naming though these mappings can be pretty pro forma. However, transformations can 
+also be applied to input from the user interface. 
+
+In the example below, the user-interface accepts the names of assemblies from the user and transforms them into object
+references before passing them on to the method. (This prevents a race condition from occurring if multiple apps are 
+writing to the same object name or if the object is renamed) In the output mapping, the the output (a single object in
+this example) is unpacked into target properties. These output properties are used to visualize the result of the app 
+(thus the need to return information about the report object).
+```
+"behavior" : {
+    "service-mapping": {
+            "url": "",
+            "name": "kb_quast",
+            "method": "run_QUAST_app",
+            "input_mapping": [
+                {
+                    "narrative_system_variable": "workspace",
+                    "target_property": "workspace_name"
+                },
+                {
+                    "input_parameter": "assemblies",
+                    "target_type_transform": "list<ref>",
+                    "target_property": "assemblies"
+                }
+            ],
+            "output_mapping": [
+                {
+                    "service_method_output_path": [0,"report_name"],
+                    "target_property": "report_name"
+                },
+                {
+                    "service_method_output_path": [0,"report_ref"],
+                    "target_property": "report_ref"
+                },
+                {
+                    "constant_value": "5",
+                    "target_property": "report_window_line_height"
+                },
+                {
+                    "service_method_output_path": [0],
+                    "target_property": "QUAST_result"
+                },
+                {
+                    "input_parameter": "assemblies",
+                    "target_property": "input_assemblies"
+                },
+                {
+                    "narrative_system_variable": "workspace",
+                    "target_property": "workspace_name"
+                }
+            ]
+        }
+    }
+```
+
```

Comment:
An example of the transformations might be useful here, too. So, the cell inputs look like one thing, the structure passed to the app looks like another. Likewise with the results - the actual function the app runs returns one specific structure, and that output mapping is used to turn it into another that the viewers can use. Having a concrete example here of what that transform looks like would be nice.